### PR TITLE
Add type annotations to conv-relu

### LINKED ISSUE #47679: Enable torch.nn.intrinsic.quantized.modules.conv_relu typechecks during CI
`TestTypeHints.test_run_mypy` should pass with `[mypy-torch.nn.intrinsic.quantized.modules.conv_relu]` deleted from mypy.ini

cc @ezyang @malfet @rgommers @xuzhao9 @gramster

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Fixes #47679

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -131,18 +131,12 @@ ignore_errors = True
...
-ignore_errors = True
-
 [mypy-torch._appdirs]
 ignore_errors = True
 
 [mypy-torch._utils]
 ignore_errors = True
 
-[mypy-torch._overrides]
-ignore_errors = True
```
Comment (rgommers) [inline mypy.ini:144]: You removed this in another PR as well, that'll give a merge conflict. Would be best to remove this I think, and also double check that you didn't do `_lobpcg` in another PR.

**Review by rgommers — COMMENTED** [review]
Thanks @guilhermeleobas. Adding `type: ignore`'s here seems fine, please just double check the comment on `mypy.ini` changes.

**Comment by rgommers** [comment]
CI failures are unrelated.

File: mypy.ini (hunk 1 above)
Comment (guilhermeleobas) [reply mypy.ini:144]: Thanks for the review, Ralf. I've checked the other PRs I have opened and none of them touched `_lobpcg`

**Review by rgommers — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**